### PR TITLE
Implementing search

### PR DESCRIPTION
**Pull request by shumalll**

### CODE REVIEW

**Comment by tianyan-zhu** [comment]
Let me know what you __HATE__ about the search result page styling, and I will fix them. ;) 

Also seems like if you are already on the search page, and do another search, we need to do a force refresh to get the content. Will investigate and change tomorrow. 

Closes #80 #81 #110

File: packages/client/src/components/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -99,16 +135,16 @@ export class SearchBar extends React.Component<{}, SearchBarState> {
...
 
     private renderSearchResults() {
-        const { results } = this.state;
+        const results: SearchResultEntry[] = this.state.results;
         if (this.state.resultsOpen && _.isArray(results) && !_.isEmpty(results)) {
             return (
                 <div className="results-container">
                     <ul className="results-list">
                         {results.map((result) => (
                             // temporarily using result as key. Change to some id later
```
Comment (shizuko-akamoto) [inline packages/client/src/components/SearchBar/SearchBar.tsx:144]: Remove TODO

File: packages/client/src/components/TrecipeCard/TrecipeCard.tsx [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
+import { showModal } from '../../redux/Modal/action';
+import TrecipePopup, { TrecipePopupType } from '../TrecipePopup/TrecipePopup';
+import Trecipe from '../../../../shared/models/trecipe';
+import { baseURL } from '../../api';
 
 type TCProps = Trecipe & ReturnType<typeof mapDispatchToProps>;
 
 class TrecipeCard extends React.Component<TCProps> {
+    public static defaultProps: Partial<TCProps> = {
+        isReadOnly: false, // should change to true later
```
Comment (shizuko-akamoto) [inline packages/client/src/components/TrecipeCard/TrecipeCard.tsx:23]: So we have read only cards by default?

File: packages/client/src/pages/Map/DestinationCard/DestinationCard.tsx [hunk 3]
```diff
@@ -1,11 +1,21 @@
 import React from 'react';
 import { CardMenu } from '../../../components/CardMenu/CardMenu';
 import { RatingBar } from '../../../components/Rating/RatingBar';
-import { DCProps } from '../../Trecipe/DestinationCard/DestinationCard';
+// import { DCProps } from '../../Trecipe/DestinationCard/DestinationCard';
```
Comment (shizuko-akamoto) [inline packages/client/src/pages/Map/DestinationCard/DestinationCard.tsx:4]: Clean up

File: packages/client/src/services/searchService.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
+import API from '../api';
+import { AxiosResponse } from 'axios';
+import Destination from '../../../shared/models/destination';
+import User from 'server/server/api/user/user.interface';
```
Comment (shizuko-akamoto) [inline packages/client/src/services/searchService.ts:4]: User unused

File: packages/client/src/pages/SearchResult/SearchResult.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import TrecipeCard from '../../components/TrecipeCard/TrecipeCard';
+import { RootState } from '../../redux';
+import { connect } from 'react-redux';
+import { bindActionCreators, Dispatch } from 'redux';
+import { Link, withRouter, RouteComponentProps } from 'react-router-dom';
+import { fetchResultRequest } from '../../redux/SearchResult/action';
+import Trecipe from '../../../../shared/models/trecipe';
+import Destination from '../../../../shared/models/destination';
+import { DestinationCard } from '../Map/DestinationCard/DestinationCard';
+import { DestinationsState } from '../../redux/Destinations/types';
```
Comment (shizuko-akamoto) [inline packages/client/src/pages/SearchResult/SearchResult.tsx:12]: Unused import

File: packages/client/src/components/SearchBar/SearchBar.tsx [hunk 6]
```diff
@@ -67,26 +77,52 @@ export class SearchBar extends React.Component<{}, SearchBarState> {
...
-                result = ['Place1', 'Place2', 'Place3'];
+                searchService.performDestinationSearchGoogle(searchKey).then((searchResult) => {
+                    result = searchResult.map((value) => {
+                        return {
+                            name: value.name,
+                            // TODO implement destination page so we could pass props to generate a new page.
+                            href: 'placeholder',
+                            info:
+                                typeof value.formattedAddress === 'undefined'
+                                    ? 'N/A'
```
[the remark this answers is not in the document]
Comment (tianyan-zhu) [reply packages/client/src/components/SearchBar/SearchBar.tsx:116]: @shumalll

File: packages/client/src/components/TrecipeCard/TrecipeCard.tsx (hunk 2 above)
Comment (tianyan-zhu) [reply packages/client/src/components/TrecipeCard/TrecipeCard.tsx:23]: changed to true :)

File: packages/shared/models/trecipe.ts [hunk 7]
```diff
@@ -18,6 +18,7 @@ export default interface Trecipe {
     name: string;
     description: string;
     owner: string;
+    isReadOnly?: boolean;
```
[the remark this answers is not in the document]
Comment (tianyan-zhu) [reply packages/shared/models/trecipe.ts:21]: Do you want a separate props?

File: packages/client/src/components/SearchBar/SearchBar.tsx [hunk 8]
```diff
@@ -67,26 +77,52 @@ export class SearchBar extends React.Component<{}, SearchBarState> {
...
             } else {
-                result = ['Place1', 'Place2', 'Place3'];
+                searchService.performDestinationSearchGoogle(searchKey).then((searchResult) => {
+                    result = searchResult.map((value) => {
+                        return {
+                            name: value.name,
+                            // TODO implement destination page so we could pass props to generate a new page.
+                            href: 'placeholder',
+                            info:
+                                typeof value.formattedAddress === 'undefined'
```
[the remark this answers is not in the document]
Comment (tianyan-zhu) [reply packages/client/src/components/SearchBar/SearchBar.tsx:115]: @shumalll

**Comment by tianyan-zhu** [comment]
Closing. Merged changes in #122

**Pull request closed without merging**